utils/contentful.ts

```diff
@@ -81,10 +84,28 @@ async function getSynth(address: string): Promise<ContentfulSynth> {
 async function getRelatedSynths(
   synth: ContentfulSynth
 ): Promise<ContentfulSynth[]> {
-  const relatedSynths = await getSynthsByCategory(synth.category);
-  return relatedSynths.filter(
-    (relatedSynth) => synth.address !== relatedSynth.address
+  // Fetch synth in the same category
+  const relatedCmsSynths = await getSynthsByCategory(synth.category);
+  // Get the synths state to see if they're expired or not
+  const allRelatedSynths = (
+    await Promise.all(relatedCmsSynths.map(fetchCompleteSynth))
+  ).filter(errorFilter) as Synth<{ type: ContractType }>[];
+  const relevantRelatedSynths = allRelatedSynths.filter((relatedSynth) => {
+    const isExpired =
+      DateTime.now().toSeconds() > Number(relatedSynth.expirationTimestamp);
+    relatedSynth.address !== synth.address && !isExpired;
+  });
+
+  if (relevantRelatedSynths.length > 0) {
+    return relevantRelatedSynths;
+  }
+  const nextCategoryIdx =
```

Comment:
so is the idea if there are no related synths, rather than showing nothing you show the next category?

Reply:
yes, exactly!